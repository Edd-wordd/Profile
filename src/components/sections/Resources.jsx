import React from 'react'
import { Card, Grid, Container, CardActionArea, Typography } from '@mui/material'
import { CardActions, CardContent, CardMedia, Button } from '@mui/material'
import { RWebShare } from 'react-web-share'
import { useStyles } from '../styles/sections/Resources.styles'
import SectionHeader from '../../components/sections/SectionHeader'
import blogPosts from '../../data/resourceData'

function Resources(props) {
  const classes = useStyles(props)
  return (
    <div>
      <SectionHeader title="Our Resources" subTitle="We make things happen" />
      <Container maxWidth="xl">
        <Grid container direction="row" justify="center" alignItems="center">
          {blogPosts.map((card, index) => (
            <Card className={classes.root} key={index}>
              <CardActionArea>
                <CardMedia
                  className={classes.media}
                  image={card.image}
                  title="resources"
                  loading="lazy"
                />
                <CardContent>
                  <Typography gutterBottom variant="h5" component="h2">
                    {card.title}
                  </Typography>
                  <Typography variant="body2" color="textSecondary" component="p">
                    {card.description}
                  </Typography>
                </CardContent>
              </CardActionArea>
              <CardActions>
                <RWebShare
                  data={{
                    text: card.description,
                    url: card.link,
                    title: card.title,
                  }}
                  onClick={() => console.log('shared successfully')}
                >
                  <Button size="small" color="primary">
                    Share
                  </Button>
                </RWebShare>
              </CardActions>
            </Card>
          ))}
        </Grid>
      </Container>
    </div>
  )
}

export default Resources
