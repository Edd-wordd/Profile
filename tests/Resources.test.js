import React from 'react'
import testComponent from './testComponent'
import Resources from '../src/components/sections/Resources'

testComponent(<Resources />, 'Resources')
