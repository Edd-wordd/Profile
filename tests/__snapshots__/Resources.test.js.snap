// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resources should render Resources 1`] = `"Component"`;
